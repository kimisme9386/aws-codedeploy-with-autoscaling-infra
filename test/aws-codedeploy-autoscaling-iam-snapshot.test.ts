import * as s3 from '@aws-cdk/aws-s3';
import { SynthUtils } from '@aws-cdk/assert';
import { Stack } from '@aws-cdk/core';

import AwsCodedeployAutoscalingIAM from '../lib/aws-codedeploy-autoscaling-iam';

test('Create codedeploy IAM role and user', () => {
  const stack = new Stack();
  const bucket = new s3.Bucket(stack, 'CodeDeploymentBucket');

  new AwsCodedeployAutoscalingIAM(stack, 'TestCodeDeployIAM', {
    codeDeployBucketName: bucket.bucketName,
  });

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
