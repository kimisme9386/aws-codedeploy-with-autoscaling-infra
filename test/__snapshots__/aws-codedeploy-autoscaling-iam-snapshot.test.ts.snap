// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create codedeploy IAM role and user 1`] = `
Object {
  "Outputs": Object {
    "TestCodeDeployIAMTravisCIUserCodedeployApp16FE3122": Object {
      "Value": Object {
        "Ref": "TestCodeDeployIAMTravisCIUserCodedeployAppC2FB33F4",
      },
    },
  },
  "Resources": Object {
    "CodeDeploymentBucket072AA3C8": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "TestCodeDeployIAMCodeDeployEC232F3C48B": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "ec2.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonSSMManagedInstanceCore",
              ],
            ],
          },
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "s3:Get*",
                    "s3:List*",
                  ],
                  "Effect": "Allow",
                  "Resource": Array [
                    Object {
                      "Fn::Join": Array [
                        "",
                        Array [
                          "arn:aws:s3:::aws-codedeploy-",
                          Object {
                            "Ref": "AWS::Region",
                          },
                          "/*",
                        ],
                      ],
                    },
                    Object {
                      "Fn::Join": Array [
                        "",
                        Array [
                          "arn:aws:s3:::",
                          Object {
                            "Ref": "CodeDeploymentBucket072AA3C8",
                          },
                          "/*",
                        ],
                      ],
                    },
                  ],
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "CodeDeploy-EC2-Permissions",
          },
        ],
        "RoleName": "CodeDeploy-EC2",
      },
      "Type": "AWS::IAM::Role",
    },
    "TestCodeDeployIAMCodeDeployEC2InstanceProfile247C63C8": Object {
      "Properties": Object {
        "InstanceProfileName": "CodeDeploy-EC2-Instance-Profile",
        "Roles": Array [
          Object {
            "Ref": "TestCodeDeployIAMCodeDeployEC232F3C48B",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "TestCodeDeployIAMCodeDeployServiceRole7120AA57": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "codedeploy.",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ".amazonaws.com",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSCodeDeployRole",
              ],
            ],
          },
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "ec2:RunInstances",
                    "ec2:CreateTags",
                    "iam:PassRole",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "auto-scaling-group-with-launch-template",
          },
        ],
        "RoleName": "CodeDeployServiceRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "TestCodeDeployIAMPushS3BucketForCodeDeploy1AC65A3F": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:PutObject",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:aws:s3:::",
                    Object {
                      "Ref": "CodeDeploymentBucket072AA3C8",
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PushS3BucketForCodeDeploy",
        "Users": Array [
          Object {
            "Ref": "TestCodeDeployIAMTravisCIUserCodedeployAppC2FB33F4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TestCodeDeployIAMTravisCIUserCodedeployAppC2FB33F4": Object {
      "Properties": Object {
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSCodeDeployDeployerAccess",
              ],
            ],
          },
        ],
        "UserName": "TravisCiUser-Codedeploy-App",
      },
      "Type": "AWS::IAM::User",
    },
  },
}
`;
